File: 0347-top-k-frequent-elements/0347-top-k-frequent-elements.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        unordered_map<int,int>frq;
        for(int ele:nums){
            frq[ele]++;
        }
        priority_queue<pair<int,int>>pq;
        for(auto const &q:frq){
            pq.push({q.second,q.first});
        }
        vector<int>ans;
        while(!pq.empty() && k!=0){
            auto const pair=pq.top();
            ans.push_back(pair.second);
            pq.pop();
            k--;
        }
        return ans;
    }
};

File: 0347-top-k-frequent-elements/0347-top-k-frequent-elements.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        freq = {}
        for ele in nums:
            freq[ele] = freq.get(ele, 0) + 1

        sorted_freq = dict(
            sorted(freq.items(), key=lambda x: x[1], reverse=True)
        )

        res = []
        c = 0

        for ele in sorted_freq:
            if c == k:
                break

            res.append(ele)
            c += 1

        return res